### DISCUSSION TITLE
lldpctl_get_interfaces fails after a couple of hours

### DISCUSSION THREAD

**Discussion by HectorFire** [lldpd/lldpd#610] [Q&A]
Hi Vicent

I am implementing a code in which I am looking for new neighbors each 8 seconds, in my function I am using the API lldpctl_get_interfaces.
The application is running properly at the beginning, but after a cople of hours this API returns NULL and reports the error "Unable to connect to lldpd daemon" and this occurs until I restarts the application.
Do you have any idea what could be happen?

Thanks for your support.

**Comment by HectorFire** [comment]
I freed the connection using the API  lldpctl_release. I think this solved the issue.

**Comment by A-lullaby** [comment]
Hello, I would like to ask why I also use the lldpctl_get_interfaces function, but why I can not get information, my program connected to lldpd, but did not get neighbor information disconnected
